tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java

```diff
@@ -232,14 +224,18 @@ private Publication toPublicationWithPublishedFiles(Publication publication) {
     }
 
     private List<AssociatedArtifact> convertFilesToPublished(Publication publication) {
-        return publication.getAssociatedArtifacts().stream().map(this::updateFileToPublished).toList();
+        return publication.getAssociatedArtifacts().stream().map(this::openFiles).toList();
     }
 
-    private AssociatedArtifact updateFileToPublished(AssociatedArtifact artifact) {
-        if (isNotAdministrativeAgreement(artifact)) {
-            return ((File) artifact).toPublishedFile();
+    //TODO: Remove unpublishable file and logic related to it after we have migrated files
+    private AssociatedArtifact openFiles(AssociatedArtifact artifact) {
+        if (artifact instanceof PendingFile<?> file) {
+            return (AssociatedArtifact) file.approve();
+        }
+        if (artifact instanceof AdministrativeAgreement administrativeAgreement) {
+            return administrativeAgreement;
         } else {
-            return artifact;
+            return artifact instanceof File file ? file.toPublishedFile() : artifact;
```

Comment:
Is necessary cause AssociatedArtifact can be an AssociatedLink.

Reply:
Rejected should also be handled here maybe? What will happen with existing Open and Internal files?

Reply:
Riktig ja, godt poeng.